draft-ietf-tls-tls13.md

```diff
@@ -1566,7 +1566,9 @@ cipher_suites
   order of client preference. If the list contains cipher suites
   the server does not recognize, support, or wish to use, the server
   MUST ignore those cipher suites, and process the remaining ones as
-  usual. Values are defined in {{cipher-suites}}.
+  usual. Values are defined in {{cipher-suites}}. If the client is
+  attempting a PSK key establishment, it MUST only advertise the single
+  cipher suite associated with the PSK.
```

Comment:
OK I understand this better. I did not know that. Also reading the spec it's hard to see it.

From the [New Session Ticket](https://tlswg.github.io/tls13-spec/#NewSessionTicket) section:

> The value of the ticket to be used as the PSK identity. The ticket itself is an opaque label. It MAY either be a database lookup key or a self-encrypted and self-authenticated value. Section 4 of [RFC5077] describes a recommended ticket construction mechanism. 

Note here the usage of "MAY" as well as "recommended".

There is also nothing in the current draft that talks about how the server should do that: "If a PSK is present, check to see that it has the same hash. If not, do a full handshake."

From the [Pre Shared Key extension](https://tlswg.github.io/tls13-spec/#pre-shared-key-extension) section, I think this is one more mention to the "cipher suite associated with the PSK" that could be changed to something along the lines of "a cipher suite with a hash function associated with the PSK".

> Clients MUST verify that the server’s selected_identity is within the range supplied by the client, that the server selected the cipher suite associated with the PSK